### PR TITLE
libreswan: 3.2 -> 4.4

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

Updating libreswan and fixing various issues in the NixOS module
This closes #48029 and completes #111320.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via `nixosTests.libreswan`
- [x] Tested compilation of all pkgs that depend on this change
- [x] Tested execution of all binary files
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123254` at 0087da9f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/a275dcbe26a7ed3e1a1cefcbab2d495f">libreswan</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by rnhmjoj** [comment]
Seems aarch64 is broken upstream, I pushed a patch.

File: pkgs/tools/networking/libreswan/default.nix [hunk 1]
```diff
@@ -1,101 +1,151 @@
...
+    iproute2 iptables procps
+    coreutils gnused gawk
+    nss.tools which
   ];
 in
 
-assert docs -> xmlto != null;
 assert stdenv.isLinux -> libselinux != null;
 
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/libreswan/default.nix:46]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/networking/libreswan/default.nix [hunk 2]
```diff
@@ -1,101 +1,151 @@
...
+    gnused gawk gmp unbound pam libevent
+    libcap_ng curl nspr nss ldns
+    # needed to patch shebangs
+    python3 bash
+  ] ++ lib.optional stdenv.isLinux libselinux;
+
+  patches = [
+    # Fix compilation on aarch64, remove on next update
+    (fetchpatch {
+      url = "https://github.com/libreswan/libreswan/pull/438.patch";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/libreswan/default.nix:78]: Please fetch a commit and not a closed PR.

File: pkgs/tools/networking/libreswan/default.nix (hunk 1 above)
Comment (rnhmjoj) [reply pkgs/tools/networking/libreswan/default.nix:46]: Sorry, what are you suggesting here?

**Pull request merged**